kml/serdes_gpx.cpp

```diff
@@ -358,6 +368,69 @@ std::string GpxParser::BuildDescription() const
   return m_description + "\n\n" + m_comment;
 }
 
+namespace
+{
+
+std::string coordToString(double c)
+{
+  std::ostringstream ss;
+  ss.precision(8);
+  ss << c;
+  return ss.str();
+}
+
+void SaveCategoryData(Writer & writer, CategoryData const & categoryData)
+{
+  writer << "<metadata>\n";
+  if (!categoryData.m_name.empty())
+    writer << kIndent2 << "<name>" << categoryData.m_name.at(kDefaultLang) << "</name>\n";
+  if (!categoryData.m_description.empty())
+    writer << kIndent2 << "<desc>" << categoryData.m_description.at(kDefaultLang) << "</desc>\n";
+  writer << "</metadata>\n";
+}
+
+void SaveBookmarkData(Writer & writer, BookmarkData const & bookmarkData)
+{
+  auto const latLon = mercator::ToLatLon(bookmarkData.m_point);
+  writer << "<wpt lat=\"" << coordToString(latLon.m_lat) << "\" lon=\"" << coordToString(latLon.m_lon) << "\">\n";
+  if (!bookmarkData.m_name.empty())
+    writer << kIndent2 << "<name>" << bookmarkData.m_name.at(kDefaultLang) << "</name>\n";
+  if (!bookmarkData.m_description.empty())
+    writer << kIndent2 << "<cmt>" << bookmarkData.m_description.at(kDefaultLang) << "</cmt>\n";
+  writer << "</wpt>\n";
+}
+
+void SaveTrackData(Writer & writer, TrackData const & trackData)
+{
+  writer << "<trk>\n";
+  if (!trackData.m_name.empty())
+    writer << kIndent2 << "<name>" << trackData.m_name.at(kDefaultLang) << "</name>\n";
+  for (auto const & line : trackData.m_geometry.m_lines)
+  {
+    writer << kIndent2 << "<trkseg>\n";
+    for (auto const & point : line)
+    {
+      auto const [lat, lon] = mercator::ToLatLon(point);
+      writer << kIndent4 << "<trkpt lat=\"" << coordToString(lat) << "\" lon=\"" << coordToString(lon) << "\">\n";
+      writer << kIndent4 << "</trkpt>\n";
```

Comment:
`<trkpt>` can also have `<ele>` inside (maybe `<name>` also - need to check to we have it for track points), but we can have them as a one-liner. So in this case it would be easier to read file with nested tags if they are with indent?

Reply:
Added the heuristic, the only confusion is with
```
Altitude constexpr kInvalidAltitude = std::numeric_limits<Altitude>::min();
Altitude constexpr kDefaultAltitudeMeters = 0;

```

Looks like for gpx parser we have kInvalidAltitude when there is no ele data, while in kml there are usages for both, not clear how it is decided what to use.